### PR TITLE
Proposal: Fix for multilines and composite scripts listing

### PR DESCRIPTION
**Pull request by noirbizarre**
## Pull Request Check List

- [x] A news fragment is added in `news/` describing what is new.
- [x] Test cases added for changed code.

## Describe what you have changed in this PR.

While the current script listing is working fine when your command are short one-liners, it was broken for multilines scripts on `main` (where this PR was initially implemented).
Now that `dev` switched on a `rich` table display, I'm not sure whether it is improvement or not so I submit it anyway,

This PR improvement proposal was to merge the `Script` and `Description` column into a single `Description` column displaying the `help` if available and if not:
- display the sequence of calls for `composite`
- display the first line of script with a `⮨` ellipsis if it has been truncated

Also, the list is now sorted.

## Example

Here the same output for `main`, `dev` and this PR for this given `tool.pdm.scripts`:

```toml
[tool.pdm.scripts]
all = {composite = ["format", "lint", "cover"]}
test = "pytest better/ tests/"
format = { shell = """\
isort better/
black better/
""" }
doc = "sphinx-build docs docs/_build/html"
"doc:watch" = "sphinx-autobuild docs docs/_build/html"

[tool.pdm.scripts.lint]
shell = """\
  echo "Flakes"
  flake8 better
  echo "isort"
  isort --check better
  echo "black"
  black --check better
  echo "MyPy"
  mypy better
"""

[tool.pdm.scripts.cover]
help = "Run tests with coverage"
cmd = """
pytest
  --cov-report=term
  --cov=better
  --cov-report=html:reports/coverage
  --cov-report=xml:reports/coverage.xml
  --junitxml=reports/tests.xml
"""
```

### Main
Capture d’écran du 2022-06-22 13-21-27

Note: the `composite` task has been removed to be able to run the command

### Dev
Capture d’écran du 2022-06-22 13-22-03

### Proposal
Capture d’écran du 2022-06-22 13-22-53

### CODE REVIEW

**Comment by frostming** [comment]
The ⮨ ellipsis doesn't look good on my machine(M1 Mac with WezTerm, Rec Mono). It doesn't even render on Chrome.

Better to choose another character.

**Comment by noirbizarre** [comment]
Ok then let's change that. It was ⮨, here a list of alternatives (so you can test it and validate it is working before i update the PR) I see:
1. The classical ellipsis: `…`
2. Same arrow but thinner: `↵`
3. Same but a bit curvier: `⤶`
4. The fat one: `⮰`

Or if you have any other suggestion or preference.
I can render the porposal with each to choose if you want.

**Comment by frostming** [comment]
I prefer the classical ellipsis and fallback to three dots `...` on Windows cmd. 😞

**Comment by noirbizarre** [comment]
Updated with a classical ellipsis and a fallback on 3 dots for legacy windows terminals.

@frostming it's why I love open-source: you find an awesome tools, open source gives you the opportunity to make more awesome to you. Given I'm a full time developer, and Python packaging have been so painful those last year, I'm glad PDM exists solving my main Python irritation point and I even more glad to contribute to make it more awesome !!! (And BTW, I apologize in advance because I know I will be flooding this repository with pull requests for some time)

**Comment by frostming** [comment]
> (And BTW, I apologize in advance because I know I will be flooding this repository with pull requests for some time)

Oh great, what big would you like to propose? BTW, would you like an alpha release or are you using the dev version already?

File: pdm/cli/commands/run.py [hunk 1]
```diff
@@ -50,6 +50,21 @@ class Task(NamedTuple):
     def __str__(self) -> str:
         return f"<task [cyan]{self.name}[/]>"
 
+    @property
+    def short_description(self) -> str:
+        """
+        A short one line task description
+        """
+        if self.kind == "composite":
+            fallback = " ➤ ".join(self.args)
```
Comment (frostming) [inline pdm/cli/commands/run.py:59]: A fallback to `>` as well?

**Review by frostming — APPROVED** [review]

**Comment by noirbizarre** [comment]
And it's updated 👍🏼 

An alpha release would be great (I can use the dev version but I prefer a released version, easier issue tracking)

Between others, I'd like to submit PRs for:
- yarn-like root script (in progress): make scripts available as root cmd as long as they don't conflict with an existing one (aka. the `test` script should be available as `pdm test`
- a bit of documentation about possible usages with `direnv`, writing plugins...
- some plugins idea: 
  - `pdm-create`: bootstrap projects using templates (like cookie-cutter or copier but builtin, same as `yarn create`, should bring more value than a simple `pdm new = mkdir + cd + pdm init`
  - `pdm-releaser`: like `go-releaser` but for and based on `pdm`
  - `pdm-vscode`: create/update vscode workspace/directory settings automatically
- and contribute on the remaining tasks on the path to 2.0 (edtiables management, XDG dirs...)

I don't know in which order and when, yarn-like root scripts is already started so it might be the next one but after that I think I will help for the 2.0 critical path, plugins can wait

**Comment by frostming** [comment]
>and contribute on the remaining tasks on the path to 2.0 (editables management, XDG dirs...)

I've updated the meta issue #743 about the remaining tasks, FYI edtiable management has been finished.

**Pull request merged**